src/bt-core-io.c

```diff
@@ -465,27 +468,78 @@ energy_info_cb(bt_activity_energy_info *energy_info ATTRIBS(UNUSED))
   /* nothing to do */
 }
 
+struct alarm_state {
+  alarm_cb cb;
+  void* data;
+};
+
+static enum ioresult
+alarm_event_in(int fd, void* data)
+{
+  uint64_t num_expirations;
+  ssize_t res;
+
+  res = TEMP_FAILURE_RETRY(read(fd, &num_expirations, sizeof(num_expirations)));
```

Comment:
I don't think you need to read the content here. |remove_timer| will close the file descriptors anyway.